kofta/src/vscode-webview/modules/room-chat/ProfileModalFetcher.tsx

```diff
@@ -1,53 +1,68 @@
 import { useAtom } from "jotai";
-import React, { useLayoutEffect } from "react";
+import React, { useEffect, useLayoutEffect } from "react";
 import { wsend } from "../../../createWebsocket";
 import { useCurrentRoomStore } from "../../../webrtc/stores/useCurrentRoomStore";
-import { meAtom, useCurrentRoomInfo } from "../../atoms";
+import { meAtom, useCurrentRoomInfo, userProfileAtom } from "../../atoms";
 import { ProfileModal } from "../../components/ProfileModal";
+import { RoomUser } from "../../types";
 import { RoomChatMessage } from "./useRoomChatStore";
 
 interface ProfileModalFetcherProps {
-  userId: string;
-  onClose: () => void;
-  messageToBeDeleted?: RoomChatMessage | null;
+	userId: string;
+	userIdType?: "uuid" | "username";
+	onClose: () => void;
+	messageToBeDeleted?: RoomChatMessage | null;
 }
 
 export const ProfileModalFetcher: React.FC<ProfileModalFetcherProps> = ({
-  userId,
-  onClose,
-  messageToBeDeleted,
+	userId,
+	userIdType = "uuid",
+	onClose,
+	messageToBeDeleted,
 }) => {
-  const { currentRoom: room } = useCurrentRoomStore();
-  const [me] = useAtom(meAtom);
-  const { isMod: iAmMod, isCreator: iAmCreator } = useCurrentRoomInfo();
-  const profile = room?.users.find((x) => [x.id, x.username].includes(userId));
+	const { currentRoom: room } = useCurrentRoomStore();
+	const [me] = useAtom(meAtom);
+	const [userProfile] = useAtom(userProfileAtom);
+	const { isMod: iAmMod, isCreator: iAmCreator } = useCurrentRoomInfo();
 
-  useLayoutEffect(() => {
-    if (
-      profile &&
-      me &&
-      profile.id !== me.id &&
-      (profile.youAreFollowing === undefined ||
-        profile.youAreFollowing === null)
-    ) {
-      wsend({ op: "follow_info", d: { userId: profile.id } });
-    }
-  }, [me, profile]);
-  if (!room) {
-    return null;
-  }
-  if (!profile) {
-    return null;
-  }
-  return (
-    <ProfileModal
-      iAmCreator={iAmCreator}
-      iAmMod={iAmMod}
-      isMe={profile?.id === me?.id}
-      room={room}
-      onClose={onClose}
-      profile={profile}
-      messageToBeDeleted={messageToBeDeleted}
-    />
-  );
+	const profileFromRoom: RoomUser | undefined = room?.users.find((x) =>
+		[x.id, x.username].includes(userId)
+	);
+
+	const profile = profileFromRoom || userProfile;
+
+	useEffect(() => {
+		profileFromRoom ||
+			wsend({ op: "get_user_profile", d: { userId, userIdType } });
```

Comment:
this is a clunky pattern, I've switched to using `react-query` to fetch stuff see `PublicRoomsQuery` as an example  https://github.com/benawad/dogehouse/blob/staging/kofta/src/vscode-webview/pages/Home.tsx#L42

(you'll also use an f_handler for this on the backend)

Reply:
Even if I use `react-query`, I can still use broadcast events to other people right?

Reply:
yeah it just replaces wsend